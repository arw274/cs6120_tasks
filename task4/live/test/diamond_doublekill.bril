@main (input: int) {
    b: int = const one;
    cond0: bool = ge input b;
    br cond0 .ge .zero;
.ge:
    b: int = sub input one;
    jmp .end;
.zero:
    b: int = const input;
    jmp .end;
.end:
    ret;
}