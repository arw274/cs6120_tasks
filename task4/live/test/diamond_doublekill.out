Function: main

Block 0 live variables at input:
['input', 'one']
Block ge live variables at input:
['input', 'one']
Block zero live variables at input:
['input']
Block end live variables at input:
[]

